mm2src/coins/eth/wallet_connect.rs

```diff
@@ -183,15 +185,17 @@ fn extract_pubkey_from_signature(
 ) -> MmResult<(H520, Address), EthWalletConnectError> {
     let account =
         H160::from_str(&account[2..]).map_to_mm(|err| EthWalletConnectError::InternalError(err.to_string()))?;
-    let uncompressed = {
+    let uncompressed: H520 = {
         let message_hash = hash_message(message.to_string());
         let signature = Signature::from_str(&signature_str[2..])
             .map_to_mm(|err| EthWalletConnectError::InvalidSignature(err.to_string()))?;
-        recover(&signature, &message_hash).map_to_mm(|err| {
+        let pubkey = recover(&signature, &message_hash).map_to_mm(|err| {
             let error = format!("Couldn't recover a public key from the signature: '{signature:?}, error: {err:?}'");
             EthWalletConnectError::InvalidSignature(error)
-        })?
+        })?;
+        pubkey.serialize_uncompressed().into()
     };
+
     let mut public = Public::default();
     public.as_mut().copy_from_slice(&uncompressed[1..65]);
```

Comment:
ummm, looks like we skip the first 8bits of the `H520`, thus rendering it essentially an `H512`. maybe we could return a `Public = H512` instead of `PublicKey = H520` from `recover()`.

that's just an early suggestion, i will want to check what these trimmed 8bits actually for first.

Reply:
not really needed. but I'm trying to understand why we have two different pubkey types and making them into one if possible.
will dig more into it.